src/function.rs

```diff
@@ -196,11 +201,32 @@ where
         db: &dyn Database,
         input: Id,
         revision: Revision,
-    ) -> MaybeChangedAfter {
+    ) -> VerifyResult {
         let db = db.as_view::<C::DbView>();
         self.maybe_changed_after(db, input, revision)
     }
 
+    fn is_provisional_cycle_head<'db>(&'db self, db: &'db dyn Database, input: Id) -> bool {
+        self.get_memo_from_table_for(db.zalsa(), input)
+            .is_some_and(|memo| {
+                memo.cycle_heads()
+                    .is_some_and(|heads| heads.contains(&self.database_key_index(input)))
+            })
```

Comment:
Does `memo.cycle_heads` return an `Option`? Could it return a `CycleHeads` struct directly to simplify this call site?

Reply:
It returns `Option<&CycleHeads>` for the reason described above. I don't want to clone the `CycleHeads`, and in the `verified_final` case I need to return something signifying "no cycle heads" but I don't have an empty `CycleHeads` available to return a reference to.

Maybe I could have a single static const empty `CycleHeads` I return a reference to here?

Reply:
Yep, the static empty `CycleHeads` works, pushed that